esp8266/modespwifi.c

```diff
@@ -0,0 +1,223 @@
+    if (n_args == 0) {
+        struct ip_info ipinfo;
+        wifi_get_ip_info(STATION_IF, &ipinfo);
+        mp_obj_t tuple[3] = {
+            netutils_format_ipv4_addr((uint8_t *)&ipinfo.ip.addr, NETUTILS_BIG),
+            netutils_format_ipv4_addr((uint8_t *)&ipinfo.gw.addr, NETUTILS_BIG),
+            netutils_format_ipv4_addr((uint8_t *)&ipinfo.netmask.addr, NETUTILS_BIG),
+        };
+
+        return mp_obj_new_tuple(3, tuple);
+    } else {
+        wifi_station_dhcpc_stop();
+
+        mp_obj_t *items;
+        mp_obj_get_array_fixed_n(args[0], 3, &items);
+
+        struct ip_info ipinfo;
+
+        netutils_parse_ipv4_addr(items[0], (uint8_t *)&ipinfo.ip.addr, NETUTILS_BIG);
+        netutils_parse_ipv4_addr(items[1], (uint8_t *)&ipinfo.gw.addr, NETUTILS_BIG);
+        netutils_parse_ipv4_addr(items[2], (uint8_t *)&ipinfo.netmask.addr, NETUTILS_BIG);
+
+        wifi_set_ip_info(STATION_IF, &ipinfo);
+
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_wifi_ip_obj, 0, 1, esp_wifi_ip);
+
+STATIC mp_obj_t esp_wifi_connect(mp_uint_t n_args, const mp_obj_t *args) {
+    struct station_config config = {{0}};
+    mp_uint_t len;
+    const char *p;
+
+    p = mp_obj_str_get_data(args[0], &len);
+    memcpy(config.ssid, p, len);
+    p = mp_obj_str_get_data(args[1], &len);
+    memcpy(config.password, p, len);
+
+    if (n_args == 3) {
+        esp_wifi_ip(1, &args[2]);
+    } else {
+        wifi_station_dhcpc_start();
+    }
+
+    error_check(wifi_station_set_config(&config), "Cannot set STA config");
+    error_check(wifi_station_connect(), "Cannot connect to AP");
+
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_wifi_connect_obj, 2, 3, esp_wifi_connect);
+
+STATIC mp_obj_t esp_wifi_disconnect() {
+    error_check(wifi_station_disconnect(), "Cannot disconnect from AP");
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(esp_wifi_disconnect_obj, esp_wifi_disconnect);
+
+STATIC mp_obj_t esp_wifi_status() {
+    return MP_OBJ_NEW_SMALL_INT(wifi_station_get_connect_status());
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(esp_wifi_status_obj, esp_wifi_status);
+
+STATIC mp_obj_t esp_wifi_phy_mode(mp_uint_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return mp_obj_new_int(wifi_get_phy_mode());
+    } else {
+        wifi_set_phy_mode(mp_obj_get_int(args[0]));
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_wifi_phy_mode_obj, 0, 1, esp_wifi_phy_mode);
+
+STATIC mp_obj_t esp_wifi_sleep_type(mp_uint_t n_args, const mp_obj_t *args) {
+    if (n_args == 0) {
+        return mp_obj_new_int(wifi_get_sleep_type());
+    } else {
+        wifi_set_sleep_type(mp_obj_get_int(args[0]));
+        return mp_const_none;
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_wifi_sleep_type_obj, 0, 1, esp_wifi_sleep_type);
+
+STATIC mp_obj_t esp_wifi_mac(mp_uint_t n_args, const mp_obj_t *args) {
```

Comment:
cc3200 just returns raw bytes object of the MAC address.  That saves a lot of bytes not having to make a function to parse/format the address (which is not as common as formatting/parsing an IP address).  And if you really need to format it then you can write one line of Python to do it.